### PR TITLE
Update XZ to topics

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/sourcepredict/environment.yml [hunk 1]
```diff
@@ -5,3 +5,5 @@ channels:
   - bioconda
 dependencies:
   - bioconda::sourcepredict=0.5.1
+  - conda-forge::python=3.12
```
Comment (vagkaratzas) [inline modules/nf-core/sourcepredict/environment.yml]: Curious, should we print mutliple versions channels in such cases to ensure conda envs and contaienr versions are in sync for everything?

File: modules/nf-core/sourcepredict/meta.yml [hunk 2]
```diff
@@ -24,38 +24,33 @@ input:
         type: file
         description: Sink TAXID count table in csv format
         pattern: "*.csv"
-
         ontologies:
-          - edam: http://edamontology.org/format_3752 # CSV
+          - edam: http://edamontology.org/format_3752
```
Comment (vagkaratzas) [inline modules/nf-core/sourcepredict/meta.yml]: why remove the comment?

**Review by vagkaratzas — APPROVED** [review]
A comment to consider for version printing, and just put back the ` # CSV` comment in the ontologies and should be good to merge

File: modules/nf-core/sourcepredict/meta.yml (hunk 2 above)
Comment (Joon-Klaps) [reply modules/nf-core/sourcepredict/meta.yml]: That's a woopsie..

File: modules/nf-core/sourcepredict/environment.yml (hunk 1 above)
Comment (Joon-Klaps) [reply modules/nf-core/sourcepredict/environment.yml]: Yeah, the issue was that without specifying the versions, the module was broken (still is as you can see from ci, i'm guessing floating point issues). 
Ok, I'll pin these versions.

File: modules/nf-core/sourcepredict/main.nf [hunk 3]
```diff
@@ -17,6 +17,8 @@ process SOURCEPREDICT {
     output:
     tuple val(meta), path("*.sourcepredict.csv"), emit: report
     tuple val("${task.process}"), val('sourcepredict'), eval('python -c "import sourcepredict; print(sourcepredict.__version__)"'), topic: versions, emit: versions_sourcepredict
+    tuple val("${task.process}"), val('sourcepredict'), eval('python -V | sed "s/Python //g"'), topic: versions, emit: versions_python
+    tuple val("${task.process}"), val('sourcepredict'), eval('python -c "import sklearn; print(sklearn.__version__)"'), topic: versions, emit: versions_sklearn
```
Comment (vagkaratzas) [inline modules/nf-core/sourcepredict/main.nf]: The values should be `python` and `sklearn`, else it wont show properly in the snapshot:
```
               "versions_python": [
                    [
                        "SOURCEPREDICT",
                        "sourcepredict",
                        "3.12.7"
                    ]
                ],
                "versions_sklearn": [
                    [
                        "SOURCEPREDICT",
                        "sourcepredict",
                        "1.5.2"
                    ]
                ],
```
It should be:
```
                "versions_python": [
                    [
                        "SOURCEPREDICT",
                        "python",
                        "3.12.7"
                    ]
                ],
                "versions_sklearn": [
                    [
                        "SOURCEPREDICT",
                        "sklearn",
                        "1.5.2"
                    ]
                ],
```

**Review by vagkaratzas — APPROVED** [review]
Just add the # CSV back xD
